src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -341,6 +342,157 @@ void Terminal::SwitchSelectionEndpoint()
     }
 }
 
+// Method Description:
+// - selects the next/previous hyperlink, if one is available
+// Arguments:
+// - searchForward: if true, we're scanning forward towards the end of the buffer. Otherwise, we're scanning backwards towards the top.
+// Return Value:
+// - true if we found a hyperlink to select (and selected it). False otherwise.
+void Terminal::SelectHyperlink(const bool searchForward)
+{
+    if (!_markMode)
+    {
+        // This feature only works in mark mode
+        _isTargetingUrl = false;
+        return;
+    }
+
+    // 0. Useful tools/vars
+    const auto bufferSize = _activeBuffer().GetSize();
+    const auto viewportHeight = _GetMutableViewport().Height();
+
+    // extracts the next/previous hyperlink from the list of hyperlink ranges provided
+    auto extractResultFromList = [&, onUrl = _isTargetingUrl, selectionStart = _selection->start](const std::vector<interval_tree::Interval<til::point, size_t>>& list, bool forward, bool useViewportCoords) {
+        // checks if we're already on the hyperlink and if we're trying to set it to the same one
+        auto isAlreadySelected = [onUrl, selectionStartForValidation = useViewportCoords ? _ConvertToViewportCell(selectionStart) : selectionStart](const interval_tree::Interval<til::point, size_t> range) {
+            return onUrl && selectionStartForValidation == range.start;
+        };
+
+        std::optional<std::pair<til::point, til::point>> resultFromList;
+        if (list.size() == 1)
+        {
+            if (!isAlreadySelected(list.front()))
+            {
+                resultFromList = std::make_pair(list.front().start, list.front().stop);
+            }
+        }
+        else if (list.size() > 1)
+        {
+            // multiple results, find the first one in the direction we're going
+            if (forward)
+            {
+                for (const auto& range : list)
+                {
+                    if (isAlreadySelected(range))
+                    {
+                        continue;
+                    }
+                    else if (!resultFromList || range.start < resultFromList->first)
+                    {
+                        resultFromList = std::make_pair(range.start, range.stop);
+                    }
+                }
+            }
+            else
+            {
+                for (const auto& range : list)
+                {
+                    if (isAlreadySelected(range))
+                    {
+                        continue;
+                    }
+                    else if (!resultFromList || range.start > resultFromList->first)

```

Comment:
You can combine these loops by doing something like `(range.start < resultFromList->first) == forward`. This works as long as you make sure that calls to `findContained` are properly excluding the current link's coordinates.